### ISSUE TITLE
Param values fingerprint by string form, so 3e-4 and 0.0003 are different experiments

### ISSUE THREAD

**Issue by kornsour** [kornsour/run-ledger#63]
`lineage.Run.Compute` hashes param values as the strings they arrived as, so
the same hyperparameter written three ways produces three fingerprints:

```
lr="3e-4"    -> eb26d16282d557d91b3b
lr="0.0003"  -> 77e0335cc066b1a38273
lr="0.00030" -> 4771e6d266eb039f2641
```

Worse, the two clients disagree for the same input:

| client | call | sends |
|---|---|---|
| `rlctl` | `--param lr=3e-4` | `"3e-4"` (literal) |
| Python | `Run.start(params={"lr": 3e-4})` | `"0.0003"` (`str()` in `_run.py`) |

So the same experiment recorded from the CLI and from training code never
lands in the same fingerprint group.

## Why this is the highest-priority correctness issue

It is a false **negative**, and the failure mode is silence. The empty-param
bug fixed in #62 was a false positive: the ledger claimed a difference that
wasn't there, you went looking, you found nothing. Annoying, but visible.

Here, `spread` never sees a repeat and `unattributable` never fires. The
ledger simply fails to make its one claim, and nobody notices a finding that
never appeared. For a tool whose whole value is that the record can be
trusted, this is the defect most likely to give someone a confidently wrong
answer.

## Direction

Normalize param values at the boundary rather than trusting their spelling:
parse a value that is numeric and re-emit it canonically, or type params as
JSON values rather than strings so a number arrives as a number.

Either way the string form stops being load-bearing, and both clients agree
because neither one's formatting survives to the hash.

## Constraint

This changes the fingerprint input, so it is an
[ADR 0004](docs/adr/0004-fingerprint-input-is-a-versioned-contract.md) event:
it needs a version field and a documented migration, not a patch release.
Worth doing deliberately.

Note that unlike the collapse in
[ADR 0011](docs/adr/0011-empty-string-means-not-recorded.md), this one cannot
be resolved by declaring one spelling canonical after the fact — existing
records carry whichever form their client happened to send.
